odxtools/diaglayer.py

```diff
@@ -636,9 +643,73 @@ def get_comparam(
 
         return cps[0]
 
-    def get_can_receive_id(self, protocol_name: Optional[str] = None) -> Optional[int]:
+    def get_can_mtu(self, protocol: Optional[Union[str, "DiagLayer"]] = None) -> Optional[int]:
+        """Return the maximum size of CAN frames that can be
+        transmitted in bytes.
+
+        For classic CAN busses, this is basically always 8. CAN-FD can
+        send up to 64 bytes per frame.
+        """
+        com_param = self.get_comparam("CP_CANFDTxMaxDataLength", protocol=protocol)
+        if com_param is None:
+            rx_id = self.get_can_receive_id(protocol=protocol)
+            if rx_id is not None:
+                # bus is CAN. We also use 8 bytes for CAN-FD if this
+                # parameter was not specified.
+                return 8
+
+            # bus is not CAN
+            return None
+
+        val = com_param.value
+        if not isinstance(val, str):
+            return None
+
+        m = re.search("TX_DL *= *([0-9]*)", val)
+        if m:
+            return int(m.group(1))
+
+        # unexpected format of parameter value
+        return 8
+
+    def use_can_fd(self, protocol: Optional[Union[str, "DiagLayer"]] = None) -> bool:
+        return (self.get_can_mtu(protocol=protocol) or 0) > 8
+
+    def get_can_baudrate(self, protocol: Optional[Union[str, "DiagLayer"]] = None) -> Optional[int]:
```

Comment:
You are putting too much logic into a class that's already too big, maybe move those to helper class? `diag_layer.com.get_can_baudrate()`

Reply:
Python allows for multiple inheritance, maybe make use of that to split the large file into smaller ones?

Reply:
My main concern is that `odxtools/diaglayer.py` have +1000 lines of code, we barely scratched the surface of `convenience for communication parameters`, there are like dozens of those com parameters in the standard